packages/playwright-core/src/server/browserType.ts

```diff
@@ -76,14 +77,28 @@ export abstract class BrowserType extends SdkObject {
 
   async _innerLaunchWithRetries(progress: Progress, options: types.LaunchOptions, persistent: types.BrowserContextOptions | undefined, protocolLogger: types.ProtocolLogger, userDataDir?: string): Promise<Browser> {
     try {
-      return this._innerLaunch(progress, options, persistent, protocolLogger, userDataDir);
+      return await this._innerLaunch(progress, options, persistent, protocolLogger, userDataDir);
     } catch (error) {
       // @see https://github.com/microsoft/playwright/issues/5214
-      const errorMessage = typeof error === 'object' && typeof error.message === 'string' ? error.message : '';
+      const errorMessage: string = typeof error === 'object' && typeof error.message === 'string' ? error.message : '';
       if (errorMessage.includes('Inconsistency detected by ld.so')) {
         progress.log(`<restarting browser due to hitting race condition in glibc>`);
         return this._innerLaunch(progress, options, persistent, protocolLogger, userDataDir);
       }
+      if (
+        // Chromium
+        errorMessage.includes('Missing X server') ||
+        // WebKit
+        errorMessage.includes('cannot open display') ||
+        // Firefox
+        errorMessage.includes('no DISPLAY environment variable specified')
+      ) {
+        rewriteErrorMessage(error, '\n' + wrapInASCIIBox(
+            'Looks like you launched a headed browser without having a XServer running.\n' +
+          'Set either \'headless: false\' or use \'xvfb-run <your-playwright-app>\' before running Playwright'
```

Comment:
Is it possible to get here on Windows? If so the Xvfb tip might be misleading.

Reply:
there is no such a thing in Windows, so I doubt it.